.circleci/config.yml

```diff
@@ -312,13 +334,87 @@ commands:
             echo "Canceling workflow as a step resulted in failure"
             node .circleci/cancel-jobs.mjs;
 
+  rebuild-check:
```

Comment:
This allows us to short circuit the docker build if the resulting image wouldn't be all that different. We are primarily using the images to cache yarn install state, so if the lock file hasn't changed, the images would only be marginally different at best.